proto_v2/instr_v2.proto

```diff
@@ -0,0 +1,30 @@
+syntax = "proto3";
+package fd.v2;
+
+/* As a note, the accounts and account data fields are populated at a higher
+   level than the instruction. */
+message InstrEnv {
+    /* Index into the set of txn accounts defined by the TxnEnv. */
+    uint32          program_id_index = 1;
+
+    /* Indexes into the set of txn accounts defined by the TxnEnv. */
+    repeated uint32 accounts         = 2;
+
+    /* The instruction data */
+    bytes           data             = 3;
+}
+
+message InstrEffects {
+    /* Result of the instruction execution */
+    int32              result      = 1;
+
+    /* Custom error code if applicable */
+    int32              custom_err  = 2;
+
+    /* Return data from the instruction */
+    bytes              return_data = 3;
```

Comment:
I think `return_data` should be at the txn level as its retrieved from the txn context:
`txn_ctx->return_data`